group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -8574,6 +8597,36 @@ public void testHeartbeatDuringRebalanceCausesRebalanceInProgress() throws Excep
         assertEquals(Errors.REBALANCE_IN_PROGRESS.code(), heartbeatResponse.errorCode());
     }
 
+    @Test
+    public void testListGroups() {
+        String genericGroupId = "generic-group-id";
+        String consumerGroupId = "consumer-group-id";
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder().build();
+        context.updateLastWrittenOffset(context.lastWrittenOffset);
+        GenericGroup genericGroup = context.createGenericGroup(genericGroupId);
+        ConsumerGroup consumerGroup = context.createConsumerGroup(consumerGroupId);
+        context.updateLastWrittenOffset(context.lastWrittenOffset + 2);
```

Comment:
In other tests, we use `replay()` to build the state. For instance, see `testConsumerGroupStates`. I wonder if we could reuse the same pattern here in order to keep the tests homogenous. What do you think?

Should we also test the state filtering somehow? One way would be to add a new member to the consumer group at some point.

Reply:
I updated this test case based on my own understanding. Please help to see if it meets expectations.